addons/mail/static/src/discuss/core/common/thread_service_patch.js

```diff
@@ -47,4 +47,25 @@ patch(ThreadService.prototype, {
             thread.isLoadingAttachments = false;
         }
     },
+
+    async muteThreadByMinutes(thread, minutes) {
+        const muteUntil = await this.rpc("/discuss/channel/mute", {
+            channel_id: thread.id,
+            minutes,
+        });
+        this.update(thread, {
+            notification_mute_until_dt: muteUntil,
+        });
+    },
```

Comment:
I would define this signature: `muteThread(thread, { minutes = false } ={})`.
_(not fond of long name, and boolean positional param when it's optional)_